packages/xo-server-sdn-controller/src/index.js

```diff
@@ -607,6 +779,57 @@ class SDNController extends EventEmitter {
 
   // ---------------------------------------------------------------------------
 
+  async _electNewPoolCenter(crossPoolNetwork) {
+    delete crossPoolNetwork.poolCenter
+
+    const crossPoolCenter = crossPoolNetwork.pools.find(poolRef => {
+      const poolNetwork = this._getPoolNetwork(poolRef, crossPoolNetwork)
+      return poolNetwork.starCenter !== undefined
+    })
+    if (crossPoolCenter !== undefined) {
+      const poolNetwork = this._getPoolNetwork(
+        crossPoolCenter,
+        crossPoolNetwork
+      )
+      const xapi = find(this._xapis, xapi => xapi.pool.$ref === crossPoolCenter)
+      const network = xapi.getObjectByRef(poolNetwork.network)
+      const pool = xapi.getObjectByRef(crossPoolCenter)
+      crossPoolNetwork.poolCenter = crossPoolCenter
+      log.debug('New pool center in cross-pool network', {
+        network: network.name_label,
+        poolCenter: pool.name_label,
+        uuid: crossPoolNetwork.uuid,
+      })
+    } else {
+      log.error('No available pool center for cross-pool network', {
+        uuid: crossPoolNetwork.uuid,
+      })
+      return
```

Comment:
Shouldn't you throw if it fails? What's supposed to happen if you can't elect a new pool?

Reply:
No, If no pool is available it means no host of all the pools is online, everything will work again as soon a one host is back up.